crates/core/tedge_agent/src/agent.rs

```diff
@@ -371,39 +426,19 @@ impl SmAgent {
 
     async fn handle_software_list_request(
         &self,
+        persistance_store: &Journal,
```

Comment:
Seeing this `persistance_store` passed to each and every `handle` function and everything above its call stack as well, just wondering if it makes sense to make it a field of the `SmAgent` itself (as it was before), as its also an integral part of the agent's functioning.